mtd.cc

```diff
@@ -703,14 +708,14 @@ main(int argc, char *argv[])
   log_epoch_interval.tv_sec = 0;
   log_epoch_interval.tv_usec = 200000;
 
-  // increment the global epoch every second
+  // set a timer for incrementing the global epoch
   if (!dotest) {
       signal(SIGALRM, epochinc);
       struct itimerval etimer;
-      etimer.it_interval.tv_sec = 1;
-      etimer.it_interval.tv_usec = 0;
-      etimer.it_value.tv_sec = 1;
-      etimer.it_value.tv_usec = 0;
+      etimer.it_interval.tv_sec = epoch_interval_ms / 1000;
+      etimer.it_interval.tv_usec = (epoch_interval_ms % 1000) * 1000;
+      etimer.it_value.tv_sec = epoch_interval_ms / 1000;
+      etimer.it_value.tv_usec = (epoch_interval_ms % 1000) * 1000;
```

Comment:
That way we could, if needed, measure intervals that aren't a multiple of milliseconds.